aws-s3-transfer-manager/src/operation/download/handle.rs

```diff
@@ -1,34 +1,56 @@
+use std::sync::Arc;
+
+use crate::error::{self, ErrorKind};
+use tokio::{
+    sync::{oneshot::Receiver, Mutex, OnceCell},
+    task,
+};
+
 /*
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
 use crate::operation::download::body::Body;
-use crate::operation::download::object_meta::ObjectMetadata;
-use tokio::task;
 
-use super::DownloadContext;
+use super::object_meta::ObjectMetadata;
 
 /// Response type for a single download object request.
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct DownloadHandle {
-    /// Object metadata
-    pub object_meta: ObjectMetadata,
+    /// Object metadata receiver.
+    pub(crate) object_meta_receiver: Mutex<Option<Receiver<ObjectMetadata>>>,
```

Comment:
Hmm, this is the oneshot receiver (https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Receiver.html), which doesn't have the `is_closed` function (maybe we can just use mpsc to make it simpler?). The problem is that there is no `recv` function for the oneshot channel; it's just a future that we need to await. Awaiting a future moves it out of the struct, so we need something to replace it—thus the `Option`—so that we can take ownership of the receiver, and that requires mutability. 

I didn't want`object_meta` function to take a mutable reference to self because we can only have one mutable reference at a time, and the reference returned by the `object_meta()` function was linked to the mutable reference if `object_meta()` was taking `&mut self`. So, I settled on a `Mutex` with interior mutability.